airbyte-cdk/python/airbyte_cdk/sources/concurrent_source/concurrent_source.py

```diff
@@ -113,12 +117,17 @@ def _submit_initial_partition_generators(self, concurrent_stream_processor: Conc
 
     def _consume_from_queue(
         self,
-        queue: ThrottledQueue,
+        queue: Queue[QueueItemObject],
         concurrent_stream_processor: ConcurrentReadProcessor,
     ) -> Iterable[AirbyteMessage]:
         while airbyte_message_or_record_or_exception := queue.get():
+            try:
```

Comment:
The tests were very minimal on this class and adding this did not break anything. Given the urgency, I'll ask for a review right now but I'll go back and add some tests to make sure we document the behavior of this class